### DISCUSSION TITLE
ValueError: Detected multiple frequencies in the timeseries please pre-process data.

### DISCUSSION THREAD

**Discussion by KanIlya** [ourownstory/neural_prophet#1447] [Q&A - forecasting best practices]
Hey! I got ValueError: Detected multiple frequencies in the timeseries please pre-process data when trying to use NeuralProphet with my data.
What this error actually means? What's the best way to approach this error, how to fix it ?

Thnks in advance for answer

**Comment by SimonWittner** [comment]
Hi, 
this error probably means that your dataframe consists of multiple frequencies. I would recommend you to pre-process your dataframe, so that all values of the column 'ds' are equidistant. In the case of hourly datestamps, make sure you have a 'ds' and 'y' value for each hour.
